### PR TITLE
Added compression feature to add -Lce options to zfs send.

### PR DESCRIPTION
**Pull request by hessenpepper**
My first contribution to OSS.

-L and -e were added to zfs around the same time as -c, so I've put them all together.

### CODE REVIEW

File: bin/znapzend [hunk 1]
```diff
@@ -164,6 +164,15 @@ use 'pfexec' for zfs commands (e.g. when running znapzend as non-priviledged use
...
 use 'sudo' for zfs commands (e.g. when running znapzend as non-priviledged user)
 
+=item compression
+
+use 'compression' to add options -Lce to the zfs send command
+If a source and destination volume are both using compression, zfs send will, by
+default, uncompress the data before sending. Zfs recv will then compress it again before 
+writing it to disk. Using -c will skip the unnecessary uncompress-compress stages.
+This decreases CPU load on both source and destination as well as reduce network
+bandwidth. The -L and -e options were thrown in for good measure.
```
Comment (oetiker) [inline bin/znapzend]: maybe mention a key word for L and e in brackets ... for good measure is a bit very flippant I think

**Comment by oetiker** [comment]
thanks!

**Comment by hadfl** [comment]
if you'd change https://github.com/oetiker/znapzend/pull/280/files#diff-33cf5e498f84195d4308abccb7c59d70R314

to an array `my @sendOpt = $self->compressed ? qw(-Lce) : ();`

you would not need the checks `if($sendOpt){` but could just do:
`@cmd = ([@{$self->priv}, 'zfs', 'send', @sendOpt, ...`

as the empty array (in case compressed option is not set) won't introduce a member to the command array.

File: bin/znapzend (hunk 1 above)
Comment (hessenpepper) [inline bin/znapzend]: I made that change and fixed a problem with a little help from @hadfl

**Comment by oetiker** [comment]
thanks for the new feature!

**Pull request merged**